dcm4che-core/src/main/java/org/dcm4che3/io/DicomInputStream.java

```diff
@@ -760,27 +745,30 @@ private void readFragments(Attributes attrs, int fragsTag, VR vr)
     }
 
     public byte[] readValue() throws IOException {
-        int valLen = length;
+
         try {
-            if (valLen < 0)
-                throw new EOFException(); // assume InputStream length < 2 GiB
-            int allocLen = allocateLimit >= 0
-                    ? Math.min(valLen, allocateLimit)
-                    : valLen;
-            byte[] value = new byte[allocLen];
-            readFully(value, 0, allocLen);
-            while (allocLen < valLen) {
-                int newLength = Math.min(valLen, allocLen << 1);
-                value = Arrays.copyOf(value, newLength);
-                readFully(value, allocLen, newLength - allocLen);
-                allocLen = newLength;
+
+            if (length < 0) {
+                throw new EOFException("DicomInputStream length is less than zero");
+            }
+
+            if (length > allocateLimit) {
```

Comment:
Looks like the previous code allowed for allocateLimit = 0 to be interpreted as "no limit". If this continues as the expected behavior, then we might see unnecessary warnings for every alocation.

Reply:
Correction: unlimited allocation was "-1", not 0.

Reply:
OK, I'll update it to first check that the allocationLimit is non-zero.